### PR TITLE
Use the same kind node image criteria for all hack scripts

### PR DESCRIPTION
**Pull request by ferhoyos**
### Describe the Change

Currently, the Kind node image is defined in multiple hack scripts according to different criteria (sometimes it is hardcoded, sometimes the latest version is used, and sometimes it is not). This PR centralizes the criteria for determining the Kind node image to be used based on the Istio version. It also ensures that the latest Kind node image is used for the latest Istio version (e.g., multicluster scripts were installing kind 1.27 which is not supported by the latest Istio version 1.24.0)

### Steps to Test the PR

- Verify that the hack scripts use the latest Kind version with the `master` branch.
- Check that specific Kind versions are used for previous Istio versions.

### Automation Testing

N/A

### CODE REVIEW

File: hack/istio/multicluster/env.sh
Comment (jmazzitelli) [inline hack/istio/multicluster/env.sh:21]: i don't think we want to get rid of this. This is a good validation feature because it helps catch instances where the code is using a variable that was never defined (or it catches a typo in a variable name). This has helped me catch many bugs that I otherwise would have missed.

If you expect a variable may not be defined in a particular place in the code, then simply use the syntax `${VAR:-}`.

File: hack/setup-kind-in-ci.sh
Comment (jmazzitelli) [inline hack/setup-kind-in-ci.sh:372]: not sure why the double-brackets are for... `-n` test need only single bracket
```suggestion
  if [ -n "${KIND_NODE_IMAGE}" ]; then
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
One change is probably trivial (the double-bracket thing), but I think we should keep `set -u` because without it, it too easily allows for bugs to creep into these scripts.

File: hack/istio/multicluster/env.sh
Comment (ferhoyos) [reply hack/istio/multicluster/env.sh:21]: Thanks @jmazzitelli, I did not know the syntax  `${VAR:-}` for unbound variables. I have restored the `set -u` flag again.

**Comment by jmazzitelli** [comment]
We should not merge this until all CI tests are green (because this PR affects the tests themselves), and there is a failure here in the integration tests.

**Comment by ferhoyos** [comment]
Agreed. The cypress test issue is happening in all PRs, so we need to fix and rebase before approval.

**Comment by jmazzitelli** [comment]
The CI bug is fixed - rebase this PR and let's see if this all goes green

**Review by jmazzitelli — APPROVED** [review]
lgtm - all tests pass

**Pull request merged**
